execution/registry/registry.go

```diff
@@ -16,38 +16,112 @@ package registry
 
 import (
 	"fmt"
+	"net"
 
 	"github.com/hyperledger/burrow/crypto"
 )
 
-func AuthorizedPeersProvider(state IterableReader) func() ([]string, []string) {
-	return func() ([]string, []string) {
-		var peerIDs, peerAddrs []string
+type NodeStats struct {
+	Addresses map[string]map[crypto.Address]struct{}
+}
+
+func NewNodeStats() NodeStats {
+	return NodeStats{
+		Addresses: make(map[string]map[crypto.Address]struct{}),
+	}
+}
 
-		for _, node := range state.GetNodes() {
-			peerIDs = append(peerIDs, node.TendermintNodeID)
-			peerAddrs = append(peerAddrs, node.NetworkAddress)
-		}
+func (ns *NodeStats) GetAddresses(net string) []crypto.Address {
+	nodes := ns.Addresses[net]
+	addrs := make([]crypto.Address, 0, len(nodes))
+	for id := range nodes {
+		addrs = append(addrs, id)
+	}
+	return addrs
+}
+
+func (ns *NodeStats) Insert(net string, id crypto.Address) {
+	_, ok := ns.Addresses[net]
+	if !ok {
+		ns.Addresses[net] = make(map[crypto.Address]struct{})
+	}
+	ns.Addresses[net][id] = struct{}{}
```

Comment:
`Address` is very overloaded, in some places we have `NodeAddress`. Can we use `NetworkAddress` or `NetAddress` to make it clearer when we are talking about network.